thunx-predicates-solr/src/main/java/eu/xenit/contentcloud/thunx/predicates/solr/SolrConverter.java

```diff
@@ -0,0 +1,89 @@
+package eu.xenit.contentcloud.thunx.predicates.solr;
+
+import eu.xenit.contentcloud.thunx.predicates.model.*;
+
+class SolrConverter implements ThunkExpressionVisitor<String> {
+
+    @Override
+    public String visit(Scalar<?> scalar) {
+        return scalar.getValue().toString();
+    }
+
+    @Override
+    public String visit(FunctionExpression<?> function) {
+
+        // convert all the terms
+        var terms = function.getTerms().stream()
+                .map(term -> term.accept(this))
+                .toArray(String[]::new);
+
+        switch (function.getOperator()) {
+            // case of boolean expressions
+            case EQUALS:
+                if (terms.length != 2) {
+                    throw new IllegalArgumentException(String.format("Expecting function expression with 2 terms. Got %s terms", terms.length));
+                }
+
+                return String.format("%s:%s", terms[0], terms[1]);
+            case OR:
+                StringBuilder builder = new StringBuilder();
+                for (int i=0; i < terms.length; i++) {
+                    if (i > 0) {
+                        builder.append(" OR ");
+                    }
+                    builder.append(terms[i]);
+                }
+                return builder.toString();
+            default:
+                throw new UnsupportedOperationException(
+                        "Operation '" + function.getOperator() + "' not implemented");
+        }
+    }
+
+    @Override
+    public String visit(SymbolicReference symbolicReference) {
+        // TODO check that symbolic-ref 'subject' matches the PathBuilder subject ?
```

Comment:
this check is actually performed at line 48

(and probably just copied from the querydsl converter, where I left in the TODO after adding the check)

Reply:
Yes, I just coped the code and didnt clean up.  Will remove.